### PR TITLE
hwloc: updated package

### PR DESCRIPTION
**Pull request by alalazo**
This includes:

* Added latest available versions (1.11.9, 2.0.0)
* Added variants to enable graphical output from lstopo
* Enabled build of the bundled netloc package if @2.0.0:
* Added dependency from numactl if @:1.11.9

@nazavode @rmsds

### CODE REVIEW

File: var/spack/repos/builtin/packages/hwloc/package.py [hunk 1]
```diff
@@ -59,25 +65,37 @@ class Hwloc(AutotoolsPackage):
     variant('pci', default=(sys.platform != 'darwin'),
             description="Support analyzing devices on PCI bus")
     variant('shared', default=True, description="Build shared libraries")
+    variant(
+        'cairo',
+        default=False,
+        description='Enable the Cairo back-end of hwloc\'s lstopo command'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/hwloc/package.py]: Just use double quotes to wrap the string and you won't need the backslash.

**Comment by nazavode** [comment]
Looks like no `openmpi` version supports the latest `hwloc`, configure fails with:

```
configure: error: OMPI does not currently support hwloc v2 API
```

Blacklisted `hwloc@2:` with [this](https://gist.github.com/nazavode/5134631b144210f9eed7ad2a7df00d1b) and everything seems to work fine. Trying on a bunch of dockerized `arch`s asap.

**Comment by alalazo** [comment]
I also plan to have a look at `openmpi` (we are starting the updates for all the software installed on our clusters). I could either mark, for the time being, `hwloc` preferred version:
```python
version('1.11.9', '4d5f5da8b1d09731d82e865ecf3fa399', preferred=True)
```
or work on `openmpi` in this same PR + apply the patch @nazavode is suggesting. Both solutions should leave:
```console
$ spack install openmpi
```
in a working state. Let me know which one you prefer.

**Comment by adamjstewart** [comment]
I prefer the latter. There's nothing wrong with `hwloc@2`, so `hwloc` should still default to the newest version. It's a restriction due to `openmpi`, not on `hwloc` itself.

**Comment by alalazo** [comment]
Thanks! I'll be back with other packages soon!

**Comment by nazavode** [comment]
@alalazo in the meantime, confirmed working on:

- `linux-ubuntu18.04-x86_64`
- `linux-ubuntu16.04-x86_64`
- `linux-centos7-x86_64`
- `linux-centos6-x86_64`

File: var/spack/repos/builtin/packages/openmpi/package.py [hunk 2]
```diff
@@ -204,6 +206,12 @@ class Openmpi(AutotoolsPackage):
     provides('mpi@:3.1', when='@2.0.0:')
 
     depends_on('hwloc')
+    # ompi@:3.0.0 doesn't support newer hwloc releases:
+    # "configure: error: OMPI does not currently support hwloc v2 API"
+    # Future ompi releases may support it, needs to be verified.
+    # See #7483 for context.
+    depends_on('hwloc@:1.999')
```
Comment (ax3l) [inline var/spack/repos/builtin/packages/openmpi/package.py]: @alalazo I think a `when='@:3.0.0'` is missing here

**Pull request merged**